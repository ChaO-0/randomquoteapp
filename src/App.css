@import url("https://fonts.googleapis.com/css?family=Open+Sans:400italic");
body {
  transition: 1s;
}

blockquote {
  font-size: 1.4em;
  width: 40%;
  margin: 30vh auto;
  font-family: "Open Sans";
  font-style: italic;
  color: #555555;
  padding: 1.2em 30px 1.2em 75px;
  line-height: 1.6;
  position: relative;
  background: #ededed;
  border-radius: 5px;
}

blockquote::before {
  font-family: Arial;
  content: "\201C";
  font-size: 4em;
  position: absolute;
  left: 10px;
  top: -10px;
  transition: 1s;
}

blockquote::after {
  content: "";
}

blockquote span {
  display: block;
  color: #333333;
  font-style: normal;
  font-weight: bold;
  margin-top: 1em;
  text-align: right;
  transition: 0.5s;
}

a {
  text-decoration: none;
  color: white;
}

.quote {
  transition: 0.5s;
}

.fade {
  opacity: 0;
}

.new-quote {
  position: absolute;
  right: 20px;
}

.soc {
  margin: 0 4px;
  padding: 10px 15px;
}

.footer {
  margin-top: -210px;
  color: white;
  text-align: center;
}

button {
  border-style: none;
  border-radius: 3px;
  background-color: #16a085;
  padding: 10px 20px;
  color: white;
  cursor: pointer;
  outline: none;
  transition: 1s;
}

button:hover {
  opacity: 0.9;
}

@media only screen and (max-width: 768px) {
  /* For mobile phones: */
  blockquote {
    font-size: 16px;
    width: 60%;
    font-family: "Open Sans";
    font-style: italic;
    color: #555555;
    padding: 1.2em 30px 1.2em 75px;
    line-height: 1.6;
    position: relative;
    background: #ededed;
  }
}
