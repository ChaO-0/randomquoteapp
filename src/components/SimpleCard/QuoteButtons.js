import React from "react";
import { connect } from "react-redux";

const QuoteButtons = ({ quote, author, color, handleClick }) => {
  return (
    <div>
      <a
        href={`https://twitter.com/intent/tweet?hashtags=quotes&text="${quote}" -${author}`}
        target="_blank"
        rel="noopener noreferrer"
      >
        <button
          className="soc"
          style={{ background: !!color ? color : "" }}
          aria-label="Left Align"
        >
          <i className="fa fa-twitter"></i>
        </button>
      </a>
      <button
        className="new-quote"
        style={{ background: !!color ? color : "" }}
        onClick={handleClick}
        aria-label="Right Align"
      >
        New Quote
      </button>
    </div>
  );
};

const mapStateToProps = (state) => ({
  quote: state.quote.content,
  author: state.quote.author,
});

export default connect(mapStateToProps)(QuoteButtons);
export { QuoteButtons };
