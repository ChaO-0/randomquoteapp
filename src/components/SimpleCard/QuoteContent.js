import React from "react";
import { connect } from "react-redux";

const QuoteContent = ({ quote, author, error, loading, color }) => {
  return (
    <>
      <div className={loading ? "quote fade" : "quote"}>
        {quote ? quote : error ? error : ""}
      </div>
      <span
        className={loading ? "fade" : ""}
        style={{ color: !!color ? color : "" }}
      >
        -{!!author ? author : error ? error : ""}
      </span>
    </>
  );
};

const mapStateToProps = (state) => ({
  quote: state.quote.content,
  author: state.quote.author,
  error: state.error,
  loading: state.loading,
});

export default connect(mapStateToProps)(QuoteContent);
export { QuoteContent };
