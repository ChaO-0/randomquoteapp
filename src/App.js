import React, { Component } from "react";
import { Provider } from "react-redux";
import store from "./components/redux/store";
import SimpleCard from "./components/SimpleCard";
import "./App.css";

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <SimpleCard />
      </Provider>
    );
  }
}

export default App;
