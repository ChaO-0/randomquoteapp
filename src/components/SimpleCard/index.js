import React, { useState, useEffect } from "react";
import { getRandomColor } from "./getRandom";
import { fetchQuote } from "../redux/actions";
import { connect } from "react-redux";
import QuoteContent from "./QuoteContent";
import QuoteButtons from "./QuoteButtons";

const SimpleCard = ({ fetchQuote }) => {
  const [Color, setColor] = useState("");

  const setBodyColor = (color) => {
    document.body.style = `background: ${color}`;
  };

  useEffect(() => {
    const color = getRandomColor();
    setBodyColor(color);
    setColor(color);
    fetchQuote();
  }, [fetchQuote]);

  const handleClick = () => {
    const color = getRandomColor();
    setBodyColor(color);
    setColor(color);
    fetchQuote();
  };

  return (
    <>
      <blockquote style={{ color: !!Color ? Color : "" }}>
        <QuoteContent color={Color} />
        <QuoteButtons color={Color} handleClick={handleClick} />
      </blockquote>
      <div data-testid="footer" className="footer">
        Copyright&copy;ChaO
      </div>
    </>
  );
};

const mapDispatchToProps = (dispatch) => ({
  fetchQuote: () => dispatch(fetchQuote()),
});

export default connect(null, mapDispatchToProps)(SimpleCard);
