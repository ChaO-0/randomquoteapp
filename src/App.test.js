import App from "./App";
import React from "react";
import ReactDOM from "react-dom";
import { QuoteContent } from "./components/SimpleCard/QuoteContent";
import { QuoteButtons } from "./components/SimpleCard/QuoteButtons";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("Renders without crashing", () => {
  const root = document.createElement("div");
  ReactDOM.render(<App />, root);
});

it("Renders copyright correctly", () => {
  const { getByTestId } = render(<App />);
  expect(getByTestId("footer").textContent).toBe("CopyrightÂ©ChaO");
});

it("Matches snapshot", () => {
  const tree = renderer.create(<QuoteContent />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("Matches button snapshot", () => {
  const tree = renderer.create(<QuoteButtons />).toJSON();
  expect(tree).toMatchSnapshot();
});
