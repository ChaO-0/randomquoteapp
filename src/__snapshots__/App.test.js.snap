// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches button snapshot 1`] = `
<div>
  <a
    href="https://twitter.com/intent/tweet?hashtags=quotes&text=\\"undefined\\" -undefined"
    rel="noopener noreferrer"
    target="_blank"
  >
    <button
      aria-label="Left Align"
      className="soc"
      style={
        Object {
          "background": "",
        }
      }
    >
      <i
        className="fa fa-twitter"
      />
    </button>
  </a>
  <button
    aria-label="Right Align"
    className="new-quote"
    style={
      Object {
        "background": "",
      }
    }
  >
    New Quote
  </button>
</div>
`;

exports[`Matches snapshot 1`] = `
Array [
  <div
    className="quote"
  >
    
  </div>,
  <span
    className=""
    style={
      Object {
        "color": "",
      }
    }
  >
    -
    
  </span>,
]
`;
